csrc/python_frontend/fusion_record.h

```diff
@@ -1251,13 +1251,26 @@ struct TensorRecord : RecordFunctor {
       is_expand[index] = is_broadcast && has_non_broadcast_size;
     }
 
+    // skip stride_order if it's alloc_domain is in the same order as with root
+    // domain. We don't need this and we should be able to just use
+    // stride_order_, but currently alloc_domain support isn't ideal and could
+    // prevent vectorization. Adding this workaround to restore performance.
+    std::vector<int64_t> stride_order;
```

Comment:
Given `stride_order_` must be a permutation, you can simply use https://en.cppreference.com/w/cpp/algorithm/adjacent_find to check it's sorted in descending order. It's semantically easier to follow and probably leads to shorter code.